modules/tensor_mechanics/src/userobjects/TensorMechanicsPlasticIsotropicSD.C

```diff
@@ -0,0 +1,138 @@
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "TensorMechanicsPlasticIsotropicSD.h"
+
+template<>
+InputParameters validParams<TensorMechanicsPlasticIsotropicSD>()
+{
+  InputParameters params = validParams<TensorMechanicsPlasticJ2>();
+  params.addRequiredParam<Real>("b", "A constant to model the influence of pressure");
+  params.addParam<Real>("c", 0.0, "A constant to model the influence of strength differential effect");
+  params.addParam<bool>("associative", true, "Flag for flow-rule, true if not specified");
+  params.addClassDescription("IsotropicSD plasticity for pressure sensitive materials and also models the strength differential effect");
+  return params;
+}
+
+TensorMechanicsPlasticIsotropicSD::TensorMechanicsPlasticIsotropicSD(const InputParameters & parameters) :
+    TensorMechanicsPlasticJ2(parameters),
+    _b(getParam<Real>("b")),
+    _c(getParam<Real>("c")),
+    _associative(getParam<bool>("associative"))
+{
+  _a = 1.0/(_b + std::pow(1.0 / std::sqrt(27.0) - _c / 27.0, 1.0 / 3.0));
+  for (unsigned i = 0; i < 3; i++)
+    for (unsigned j = 0; j < 3; j++)
+      for (unsigned k = 0; k < 3; k++)
+        for (unsigned l = 0; l < 3; l++)
+          _h(i,j,k,l) = ((i == k) * (j == l) - 1.0 / 3.0 * (i == j) * (k == l));
+}
+
+Real
+TensorMechanicsPlasticIsotropicSD::dphi_dj2(const Real j2, const Real j3) const
+{
+  return std::pow(j2, 1.0 / 2.0) / (2 * std::pow(std::pow(j2, 3.0 / 2.0) - _c * j3, 2.0 / 3.0));
+}
+
+Real
+TensorMechanicsPlasticIsotropicSD::dphi_dj3(const Real j2, const Real j3) const
+{
+  return - _c / (3 * std::pow(std::pow(j2, 3.0 / 2.0) - _c * j3, 2.0 / 3.0));
+}
+
+Real
+TensorMechanicsPlasticIsotropicSD::dfj2_dj2(const Real j2, const Real j3) const
+{
+  return std::pow(j2, -1.0 / 2.0) / (4 * std::pow(std::pow(j2, 3.0 / 2.0) - _c * j3, 2.0 / 3.0))
+    - j2 / (2 * std::pow(std::pow(j2, 3.0 / 2.0) - _c * j3, 5.0 / 3.0));
+}
+
+Real
+TensorMechanicsPlasticIsotropicSD::dfj2_dj3(const Real j2, const Real j3) const
+{
+  return _c * std::pow(j2, 1.0 / 2.0) / (3 * std::pow(std::pow(j2, 3.0 / 2.0) - _c * j3, 5.0 / 3.0));
+}
+
+Real
+TensorMechanicsPlasticIsotropicSD::dfj3_dj2(const Real j2, const Real j3) const
+{
+  return _c * std::pow(j2, 1.0 / 2.0) / (3 * std::pow(std::pow(j2, 3.0 / 2.0) - _c * j3, 5.0 / 3.0));
+}
+
+Real
+TensorMechanicsPlasticIsotropicSD::dfj3_dj3(const Real j2, const Real j3) const
+{
+  return -_c * _c * 2.0
+    / (9 * std::pow(std::pow(j2, 3.0 / 2.0) - _c * j3, 5.0 / 3.0));
+}
+
+RankTwoTensor
+TensorMechanicsPlasticIsotropicSD::dI_sigma() const
+{
+  return RankTwoTensor(RankTwoTensor::initIdentity);
+}
+
+RankTwoTensor
+TensorMechanicsPlasticIsotropicSD::dj2_dSkl(const RankTwoTensor & stress) const
+{
+  const RankTwoTensor a;
+  const Real trace = stress.trace();
+  for (unsigned i = 0; i < 3; i++)
```

Comment:
please change to prefix increment (`++i`) in all loops. This is our de facto standard. It is never slower and sometimes faster.